pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -347,6 +347,7 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 		return fmt.Errorf("failed to parse strings %v as CIDRs: %v", cidrStrings, err)
 	}
 
+	// TODO: revisit need of needPodCIDRsUpdate with this new approach
```

Comment:
ok, I understand this function, it compares the existing assigned CIDRs to the node with the proposed ones.
If they don't match it returns true to try to update.
IT is important to mention that node.Spec.PodCIDRs is inmutable, to once set any operation to try to change them is going to fail

```
 # * spec.podCIDRs: Forbidden: node updates may not change podCIDR except from "" to valid
```

that is important for your feature too, I don't know if you want to allow users to switch the cidrs on the node, but that is not possible

Reply:
there is no point of trying the update of PodCIDR, since API will block it. lets revisit this in separate PR.